src/functions-and-arrays.js

```diff
@@ -1,122 +1,248 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
-
-
+function maxOfTwoNumbers(num1, num2) {
+  if (num1 < num2) {
+    return num2;
+  } else if (num1 > num2) {
+    return num1;
+  } else {
+    return num1;
+  }
+}
 
 // Iteration #2: Find longest word
-const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
-
-function findLongestWord() {}
-
+const words = [
+  "mystery",
+  "brother",
+  "aviator",
+  "crocodile",
+  "pearl",
+  "orchard",
+  "crackpot",
+];
 
+function findLongestWord(words) {
+  let longestWord = "";
+  if (words.length === 0) {
+    return null;
+  }
+  for (let i = 0; i < words.length; i++) {
+    if (words[i].length > longestWord.length) {
+      longestWord = words[i];
+    }
+  }
+  return longestWord;
+}
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
+function sumNumbers(numbers) {
+  let sum = 0;
 
-function sumNumbers() {}
-
+  for (let i = 0; i < numbers.length; i++) {
+    sum += numbers[i];
+  }
 
+  return sum;
+}
 
 // Iteration #3.1 Bonus:
 function sum() {}
 
-
-
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function averageNumbers(numArray) {
+  if (numArray.length === 0) {
+    return null;
+  }
+  let total = 0;
 
+  for (const sumOfNumbers of numArray) {
+    total += sumOfNumbers;
```

Comment:
In the previous iteration you created a function that sums numbers (sumNumbers), so instead of having to repeat the same code for this new function (averageNumbers) you can actually use the previous function (sumNumbers):

```
function averageNumbers(numArray) {
  if (numArray.length === 0) {
    return null;
  }
  let total = sumNumbers(numArray);
  return total / numArray.length;
}
```